src/test/crimson/test_thread_pool.cc

```diff
@@ -25,28 +29,43 @@ seastar::future<> test_accumulate(ThreadPool& tp) {
   });
 }
 
-seastar::future<> test_void_return(ThreadPool& tp) {
-  return tp.submit([=] {
+seastar::future<> test_void_return(ThreadPool* tp) {
+  return tp->submit([=] {
     std::this_thread::sleep_for(10ns);
   });
 }
 
 int main(int argc, char** argv)
 {
-  ThreadPool tp{2, 128, 0};
+  std::unique_ptr<crimson::thread::ThreadPool> tp;
   seastar::app_template app;
   return app.run(argc, argv, [&tp] {
-      return tp.start().then([&tp] {
-          return test_accumulate(tp);
-        }).then([&tp] {
-          return test_void_return(tp);
-        }).handle_exception([](auto e) {
-          std::cerr << "Error: " << e << std::endl;
-          seastar::engine().exit(1);
-        }).finally([&tp] {
-          return tp.stop();
-        });
+    std::vector<const char*> args;
+    std::string cluster;
+    std::string conf_file_list;
+    auto init_params = ceph_argparse_early_args(args,
+                                                CEPH_ENTITY_TYPE_CLIENT,
+                                                &cluster,
+                                                &conf_file_list);
+    return crimson::common::sharded_conf().start(init_params.name, cluster)
+    .then([conf_file_list] {
+      return local_conf().parse_config_files(conf_file_list);
+    }).then([&tp] {
+      tp = std::make_unique<crimson::thread::ThreadPool>(2, 128, 0);
```

Comment:
```suggestion
      return seastar::do_with(crimson::thread::ThreadPool{2, 128, 0}. [](auto tp) {
        // test with tp
```

Reply:
can't use (auto tp) here. do_with build error.  also can't use (auto& tp) here.  a delete constructor is called ThreadPool(const ThreadPool&) = delete;

Reply:
oh, i see. `ThreadPool` is not movable.